examples/applications/plot_out_of_core_classification.py

```diff
@@ -169,7 +170,9 @@ def progress(blocknum, bs, size):
         if _not_in_sphinx():
             sys.stdout.write("\r")
         print("untarring Reuters dataset...")
-        tarfile.open(archive_path, "r:gz").extractall(data_path)
+        # extractall can cause a tar slip
```

Comment:
From https://github.com/scikit-learn/scikit-learn/pull/23163#issuecomment-1104836762, this does not actually prevent the exploit:

```suggestion
```

This applies to the other diffs as well.